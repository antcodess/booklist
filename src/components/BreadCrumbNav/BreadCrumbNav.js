import React from 'react';
import { withRouter } from 'react-router-dom';

import './BreadCrumbNav.scss';

const BreadCrumbNav = ({ history, title }) => {
  return (
    <div className='breadcrumb-nav'>
      <h4 onClick={() => history.push('/')}>Home /</h4>
      <span className='current-page'>{title}</span>
    </div>
  );
};

export default withRouter(BreadCrumbNav);