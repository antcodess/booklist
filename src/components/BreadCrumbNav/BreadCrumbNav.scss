.breadcrumb-nav {
  display: flex;
  margin-left: 2px;

  h4 {
    font-size: .94rem;

    .current-page {
      font-family: "Cera GR Bold";
      color: #BCBCBC;
    }
  }

}