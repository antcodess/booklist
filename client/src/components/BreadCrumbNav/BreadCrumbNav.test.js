import React from 'react';
import { shallow } from 'enzyme';
import { BreadCrumbNav } from './BreadCrumbNav';

describe('BreadCrumbNav Component', () => {
  let wrapper;
  let mockHistory;
  let mockDispatch;
  
  beforeEach(() => {

    mockHistory = {
      push: jest.fn()
    };

    const mockProps = {
      history: mockHistory,
      title: 'This is a title'
    };

    wrapper = shallow(<BreadCrumbNav { ...mockProps } />)
  });

  it('should render BreadCrumbNav Component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should call history.push when clicked', () => {
    wrapper.find('h4').simulate('click');
    expect(mockHistory.push).toHaveBeenCalled();
  });
});