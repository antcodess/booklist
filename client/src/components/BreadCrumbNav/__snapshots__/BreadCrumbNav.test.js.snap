// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumbNav Component should render BreadCrumbNav Component 1`] = `ShallowWrapper {}`;
